### PR TITLE
Add stringtie merge module (from nanoseq modules)

### PR DESCRIPTION
**Pull request by yuukiiwa**

### CODE REVIEW

File: software/stringtie/merge/meta.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            Stringtie transcript gtf output(s).
+    - annotation_gtf:
+        type: file
+        description: |
+            Annotation gtf file.
+output:
+    - merged_gtf:
+        type: map
+        description: |
+            Merged gtf from annotation and stringtie output gtfs.
```
Comment (KevinMenden) [inline software/stringtie/merge/meta.yml:26]: I know it's obvious but we usually have a `pattern` field here as well, basically describing the file ending

File: software/stringtie/stringtie/functions.nf [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+def getPathFromList(path_list) {
+    def paths = path_list.findAll { item -> !item?.trim().isEmpty() }      // Remove empty entries
+    paths     = paths.collect { it.trim().replaceAll("^[/]+|[/]+\$", "") } // Trim whitespace and trailing slashes
+    return paths.join('/')
+}
+
+/*
+ * Function to save/publish module results
+ */
+def saveFiles(Map args) {
```
Comment (KevinMenden) [inline software/stringtie/stringtie/functions.nf:41]: There's a new version of this file available in the template:
https://github.com/nf-core/tools/blob/master/nf_core/module-template/software/functions.nf

**Review by KevinMenden — COMMENTED** [review]
Awesome, thank you!
Looks really good, except the two small things that I mentioned.

File: software/stringtie/merge/main.nf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+process STRINGTIE_MERGE {
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['']) }
+
+    // Note: 2.7X indices incompatible with AWS iGenomes.
+    conda     (params.enable_conda ? "bioconda::stringtie=2.1.4" : null)
+    container "quay.io/biocontainers/stringtie:2.1.4--h7e0af3c_0"
```
Comment (KevinMenden) [inline software/stringtie/merge/main.nf:15]: Didn't catch it the first time but this is missing the singularity container, should have this structure:

```
    conda (params.enable_conda ? "bioconda::bcftools=1.11" : null)
    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
        container "https://depot.galaxyproject.org/singularity/bcftools:1.11--h7c999a4_0"
    } else {
        container "quay.io/biocontainers/bcftools:1.11--h7c999a4_0"
    }
```

**Comment by yuukiiwa** [comment]
Thanks for the comments, @KevinMenden! I will update you once I am done!

File: software/stringtie/merge/meta.yml (hunk 1 above)
Comment (yuukiiwa) [reply software/stringtie/merge/meta.yml]: I have added the patterns

File: software/stringtie/stringtie/functions.nf (hunk 2 above)
Comment (yuukiiwa) [reply software/stringtie/stringtie/functions.nf]: `functions.nf` is now using the new template

File: software/stringtie/merge/main.nf (hunk 3 above)
Comment (yuukiiwa) [reply software/stringtie/merge/main.nf]: I have added the singularity container for `stringtie2`

**Comment by yuukiiwa** [comment]
Hi @KevinMenden, thank you for your suggestions! I have addressed all of them!

**Review by KevinMenden — APPROVED** [review]
Thanks, looks good now! :+1:

**Comment by yuukiiwa** [comment]
Thank you!!

File: software/stringtie/merge/main.nf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+params.options = [:]
+def options    = initOptions(params.options)
+
+process STRINGTIE_MERGE {
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['']) }
+
+    // Note: 2.7X indices incompatible with AWS iGenomes.
```
Comment (drpatelh) [inline software/stringtie/merge/main.nf:13]: Can remove this line because this is only for STAR indices?

**Pull request merged**